pkgs/desktops/deepin/core/deepin-kwin/default.nix

```diff
@@ -0,0 +1,118 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, pkg-config
+, wrapQtAppsHook
+, qtbase
+, qttools
+, qtx11extras
+, wayland
+, dwayland
+, extra-cmake-modules
+, gsettings-qt
+, libepoxy
+, kconfig
+, kconfigwidgets
+, kcoreaddons
+, kcrash
+, kdbusaddons
+, kiconthemes
+, kglobalaccel
+, kidletime
+, knotifications
+, kpackage
+, plasma-framework
+, kcmutils
+, knewstuff
+, kdecoration
+, kscreenlocker
+, valgrind
+, breeze-qt5
+, libinput
+, mesa
+, lcms2
+, xorg
+}:
+
+stdenv.mkDerivation rec {
+  pname = "deepin-kwin";
+  version = "5.24.3-deepin.1.9";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
+    # Upstream do not set tags for a long time
+    # Use commit update version for debian/changelog
+    rev = "6bb381c00deb7427ccb319b5bbc2aeb3290b9c51";
+    sha256 = "sha256-DEOSXd+BvCv286KyTsNYlz/1yu86phN4irJNucnj7vk=";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "disable_dde-dock_preview_notify";
+      url = "https://github.com/linuxdeepin/deepin-kwin/commit/a874f798b5c3118f3f62d7dd9dfc0322d3cb88ef.patch";
+      sha256 = "sha256-ANcQ5HU52A78gOkROfiGOuYSdz+0NOm9KYkkeGLw660=";
+    })
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    extra-cmake-modules
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    qtbase
+    qttools
+    qtx11extras
+    wayland
+    dwayland
+    libepoxy
+    gsettings-qt
+
+    kconfig
+    kconfigwidgets
+    kcoreaddons
+    kcrash
+    kdbusaddons
+    kiconthemes
+
+    kglobalaccel
+    kidletime
+    knotifications
+    kpackage
+    plasma-framework
+    kcmutils
+    knewstuff
+    kdecoration
+    kscreenlocker
+
+    valgrind
+    breeze-qt5
+    libinput
+    mesa
+    lcms2
+
+    xorg.libxcb
+    xorg.libXdmcp
+    xorg.libXcursor
+    xorg.xcbutilcursor
+    xorg.libXtst
+  ];
+
+  cmakeFlags = [
+    "-DKWIN_BUILD_RUNNERS=OFF"
+    "-DDEFINE_DDE_DOCK_PATH=/run/current-system/sw/bin"
```

Comment:
What's this for?

Reply:
https://github.com/linuxdeepin/deepin-kwin/blob/86a4db4cc1cf82c92719c2644086ac4110df4c99/src/effects/screenshot/screenshotdbusinterface1.cpp#L513
Used for blocking the notification when dde-dock calling the screenshot interface，the preview function of  dde-dock will frequently call this interface

Reply:
Shall we hardcode the path to dde-dock.

Reply:
For security reasons, it should not search from `PATH`

Reply:
I mean we should just do `-DDEFINE_DDE_DOCK_PATH=${dde-dock}/bin`

Reply:
It seems that there has a problem in upstream，cmdline for 'dde-dock' is `dde-dock -r`,   not an absolute path(started by startdde)